packages/integration_test/lib/common.dart

```diff
@@ -7,17 +7,23 @@ import 'dart:convert';
 
 /// A callback to use with [integrationDriver].
 ///
-/// The callback receives the name of screenshot passed to `binding.takeScreenshot(<name>)` and
-/// a PNG byte buffer.
+/// The callback receives the name of screenshot passed to `binding.takeScreenshot(<name>)`,
+/// a PNG byte buffer representing the screenshot, and an optional Map of arguments.
 ///
 /// The callback returns `true` if the test passes or `false` otherwise.
 ///
 /// You can use this callback to store the bytes locally in a file or upload them to a service
 /// that compares the image against a gold or baseline version.
 ///
+/// The optional Map of arguments can be passed from the
+/// `binding.takeScreenshot(<name>, <args>)` callsite in the integration test,
+/// and then the arguments can be used in the `onScreenshot` handler that is defined by
+/// the Flutter driver. This Map should contain primitive types that are able to
+/// be serialized from the browser to the Flutter driver.
```

Comment:
Still not clear what values are supported. I believe we're using JSON (it's run through `Response.toJson`), so I think it's safe to say that `args` only supports values that can be serialized to JSON.